test/HealthChecks.AzureServiceBus.Tests/AzureServiceBusQueueHealthCheckTests.cs

```diff
@@ -0,0 +1,402 @@
+        }
+        else
+        {
+            _clientProvider
+                .Received(1)
+                .CreateManagementClient(ConnectionString);
+        }
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public async Task can_create_client_with_fully_qualified_endpoint(bool peakMode)
+    {
+        using var tokenSource = new CancellationTokenSource();
+        var options = new AzureServiceBusQueueHealthCheckOptions(QueueName)
+        {
+            FullyQualifiedNamespace = FullyQualifiedName,
+            Credential = _tokenCredential,
+            UsePeekMode = peakMode,
+        };
+        var healthCheck = new AzureServiceBusQueueHealthCheck(options, _clientProvider);
+        var context = new HealthCheckContext
+        {
+            Registration = new HealthCheckRegistration(HEALTH_CHECK_NAME, healthCheck, HealthStatus.Unhealthy, null)
+        };
+
+        var actual = await healthCheck
+            .CheckHealthAsync(context, tokenSource.Token)
+            .ConfigureAwait(false);
+
+        if (peakMode)
+        {
+            _clientProvider
+                .Received(1)
+                .CreateClient(options.FullyQualifiedNamespace, options.Credential);
+        }
+        else
+        {
+            _clientProvider
+                .Received(1)
+                .CreateManagementClient(options.FullyQualifiedNamespace, options.Credential);
+        }
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public async Task reuses_existing_client_when_checking_different_queue_in_same_servicebus(bool peakMode)
+    {
+        // First call
+        using var tokenSource = new CancellationTokenSource();
+        var options = new AzureServiceBusQueueHealthCheckOptions(QueueName)
+        {
+            FullyQualifiedNamespace = FullyQualifiedName,
+            Credential = _tokenCredential,
+            UsePeekMode = peakMode,
+        };
+        var healthCheck = new AzureServiceBusQueueHealthCheck(options, _clientProvider);
+        var context = new HealthCheckContext
+        {
+            Registration = new HealthCheckRegistration(HEALTH_CHECK_NAME, healthCheck, HealthStatus.Unhealthy, null)
+        };
+
+        var actual = await healthCheck
+            .CheckHealthAsync(context, tokenSource.Token)
+            .ConfigureAwait(false);
+
+        // Second call
+        var otherQueueName = Guid.NewGuid().ToString();
+        var otherOptions = new AzureServiceBusQueueHealthCheckOptions(otherQueueName)
+        {
+            FullyQualifiedNamespace = FullyQualifiedName,
+            Credential = _tokenCredential,
+        };
+        var otherHealthCheck = new AzureServiceBusQueueHealthCheck(options, _clientProvider);
+        var otherContext = new HealthCheckContext
+        {
+            Registration = new HealthCheckRegistration(HEALTH_CHECK_NAME, healthCheck, HealthStatus.Unhealthy, null)
+        };
+
+        var otherActual = await healthCheck
```

Comment:
```suggestion
        var otherActual = await otherHealthCheck
```

Please, look through test files one more time...